.github/workflows/tests.yaml

```diff
@@ -23,19 +23,26 @@ jobs:
         # At present, these images are newer than "latest". We use them to test
         # against more recent tooling versions.
         # https://github.com/actions/runner-images
-        os: [ubuntu-22.04, macos-12]
+        os: [ubuntu-22.04, macos-14]
         build_mode: [fastbuild, opt]
-        # The clang-tidy config doesn't work on macos (missing `truncate`).
         include:
+          # The clang-tidy config doesn't work on macos (missing `truncate`).
           - os: 'ubuntu-22.04'
             build_mode: 'clang-tidy'
+          # On push (post-submit), also run on `macos-12` to get Intel macOS
+          # coverage. This is less and less critical so we skip it on the PR and
+          # merge queue path by switching to a redundant OS.
```

Comment:
I think it's nice to keep for now. If nothing else, the fact that so much GitHub action infra is still using Intel macOS seems like a good reason to avoid letting it break just yet.